src/wms/src/main/java/org/geoserver/wms/legendgraphic/BufferedImageLegendGraphicBuilder.java

```diff
@@ -126,8 +133,30 @@ public BufferedImage buildLegendGraphic(GetLegendGraphicRequest request)
         // a layer group is given)
         setup(request);
         List<RenderedImage> layersImages = new ArrayList<>();
+
+        Catalog catalog = (Catalog) GeoServerExtensions.bean("catalog");
```

Comment:
instead of doing so you can simply retrieve the Locale already set in the [GetLegendGraphicRequest](https://github.com/geoserver/geoserver/blob/de184b2f9ed9a844b5291dd163e772132a07689b/src/wms/src/main/java/org/geoserver/wms/GetLegendGraphicRequest.java#L582) find the featureTypeInfo from catalog and apply the toString(Locale) on that one. 
Since the GrowableInternationalString toString method is returning always a result, I would also add a check that the requested locale is actually present among the ones hold by the internationalTitle and if not use the GeoServerDefaultLocale.

Reply:
done